src/authenticator.rs

```diff
@@ -1335,6 +1347,61 @@ where
         }
         return Ok(());
     }
+
+    fn yk_hmac<const R: usize>(&mut self, req: YKGetHMAC, reply: &mut Data<{ R }>) -> Result {
+        // Get HMAC slot command
+
+        let credential = self
+            .load_credential(req.get_credential_label()?)
+            .ok_or(Status::NotFound)?;
+        let key: &[u8] = &credential.secret;
+
+        // Make sure the set Credential is the right kind
+        ensure(
+            credential.kind == Kind::Hmac,
+            Status::IncorrectDataParameter,
+        )?;
+
+        let signature = hmac_challenge(&mut self.trussed, Algorithm::Sha1, req.challenge, key)?;
+
+        // TODO remove this check?
+        const HMAC_SHA1_LENGTH: usize = 20;
+        ensure(
+            signature.len() == HMAC_SHA1_LENGTH,
+            UnspecifiedNonpersistentExecutionError,
+        )?;
+
+        reply
+            .extend_from_slice(signature.as_slice())
+            .map_err(|_| UnspecifiedNonpersistentExecutionError)?;
+        Ok(())
+    }
+
+    fn yk_status<const R: usize>(&self, reply: &mut Data<{ R }>) -> Result {
+        // Get 6 bytes status; 3 bytes version, 3 bytes other data
+        // TODO Discuss, should this be application or runner firmware version
+        let v = OathVersion::default();
+        let firmware_version = &[v.major, v.minor, v.patch];
+        reply
+            .extend_from_slice(firmware_version)
+            .map_err(|_| UnspecifiedPersistentExecutionError)?;
+
+        // Add filler to match the expected 6 bytes
+        // TODO Check the actual data format for the YK request
+        let other_data = &[0x42, 0x42, 0x42];
+        reply
+            .extend_from_slice(other_data)
+            .map_err(|_| UnspecifiedPersistentExecutionError)?;
```

Comment:
nit: I think the `extend_from_slice` failures  should lead to `NotEnoughMemory` rather than unspecified failures.